modules/core/src/main/scala/playground/smithyutil/TransitiveCompiler.scala

```diff
@@ -25,11 +25,10 @@ final class TransitiveCompiler(
       case p @ PrimitiveSchema(_, _, _)      => underlying(p)
       case u @ UnionSchema(_, _, _, _) =>
         underlying(u.copy(alternatives = u.alternatives.map(_.mapK(this))))
-      case BijectionSchema(s, bijection) => BijectionSchema(this(s), bijection)
-      case LazySchema(suspend)           => LazySchema(suspend.map(this.apply))
-      case RefinementSchema(underlying, refinement) =>
-        RefinementSchema(this(underlying), refinement)
-      case c @ CollectionSchema(_, _, _, _) => c.copy(member = this(c.member))
+      case BijectionSchema(s, bijection)    => underlying(BijectionSchema(this(s), bijection))
+      case LazySchema(suspend)              => underlying(LazySchema(suspend.map(this.apply)))
+      case RefinementSchema(s, refinement)  => underlying(RefinementSchema(this(s), refinement))
```

Comment:
**thought:** Notably, these four will now be visited on the top-level themselves, even though they have no hints. This should make `TransitiveCompiler` more useful in general.